torchrl/modules/models/decision_transformer.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import importlib
+
+import torch
+import torch.nn as nn
+import transformers
+from transformers.models.gpt2.modeling_gpt2 import GPT2Model
+
+_has_transformers = importlib.util.find_spec("transformers") is not None
+
+
+class DecisionTransformer(nn.Module):
+    """Online Decion Transformer.
+
+    Desdescribed in https://arxiv.org/abs/2202.05607 .
+
+    Args:
+        state_dim (int): dimension of the state space
+        action_dim (int): dimension of the action space
+        config (dict): transformer architecture configuration, used to create the GPT2Config from transformers.
+
+
+    Example:
+        >>> config = {
+        ...     "n_embd": 256,
+        ...     "n_layer": 4,
+        ...     "n_head": 4,
+        ...     "n_inner": 1024,
+        ...     "activation": "relu",
+        ...     "n_positions": 1024,
+        ...     "resid_pdrop": 0.1,
+        ...     "attn_pdrop": 0.1,
+        ... }
+        >>> model = DecisionTransformer(state_dim=4, action_dim=2, config=config)
+        >>> observation = torch.randn(32, 10, 4)
+        >>> action = torch.randn(32, 10, 2)
+        >>> return_to_go = torch.randn(32, 10, 1)
+        >>> output = model(observation, action, return_to_go)
+        >>> output.shape
+        torch.Size([32, 10, 256])
+
+    """
+
+    def __init__(
+        self,
+        state_dim,
+        action_dim,
+        config,
+    ):
+        if not _has_transformers:
+            raise ImportError(
+                "transformers is not installed. Please install it with `pip install transformers`."
+            )
+        super(DecisionTransformer, self).__init__()
+
+        gpt_config = transformers.GPT2Config(
+            n_embd=config["n_embd"],
+            n_layer=config["n_layer"],
+            n_head=config["n_head"],
+            n_inner=config["n_inner"],
+            activation_function=config["activation"],
+            n_positions=config["n_positions"],
+            resid_pdrop=config["resid_pdrop"],
+            attn_pdrop=config["attn_pdrop"],
+            vocab_size=1,
```

Comment:
maybe `config.get("n_embd", some_default)`?

Reply:
```
class DecisionTransformer:
    @classmethod
    def default_config_smth(cls):
        return {...}
    @classmethod
    def default_config_smthelse(cls):
        return {...}

```

Reply:
that needs to be documented

Reply:
```
cfg = DecisionTransformer.default_config_smth()
cfg['param'] = 10
model = DecisionTransformer(..., cfg)
```

Reply:
I used a default config which gets updated if the user provides one. Let me know what you think